# Introduction-to-RNA-Seq
## Objective
This is the basics to RNA-Seq. I am utilizing multiple resources to go through the basics and hope to analyze data from papers and datasets in the future.

# DIY transcriptomics Overview
This course is by Daniel Beiting from UPenn. All [lecture material](https://diytranscriptomics.com/) and resources linked on his website. Some notes provided may stem from personal professional experience.

## Course objective
- To analyze your own bulk RNA-Seq or scRNA-Seq data
- To make a lightweight and reusable pipeline for different datas and potentially other types of assays such as ATAC-Seq
- To learn best practices for working in R/bioconductor
- To become a better data scientist
- To learn basic building blocks to create publication quality graphics
- To learn to report the owrk in a transparent and reproducible fashion

## The approach
- Open-source material (free!)
- Modular learning
- Using your personal laptop as the workstation
- RNA-Seq focused, but applicable to other genomic sequencing data
- Multiple 'on-ramps' to wrangling data into your pipeline or analysis
- Virtual lectures that are accessible at your own pace and hands-on labs to develop skills

## General steps of analysis
1. Data import
2. Data wrangling
3. Data exploration
4. Accessing public data
5. Feature Selection
6. Finding modules
7. Funcitonal enrichment analysis
8. Dynamic Reports

# Lecture 1. Introduction to RNA-Seq technology and data
## Learning objectives
- Get familiar with terminology of RNASeq/HTS (paired-end, single-end, cluster density, FastQ, index, etc).
- Understand Illumina's "Sequencing by synthesis" (SBS) technology.
- Set priorities when planning a sequencing experiment.
- Understand the basics of library prep for HTS.

## Reading Material
### [NGS technology and file formats](https://learn.gencore.bio.nyu.edu/ngs-file-formats/)
![File format flow chart](./Relative%20links/FileFormatFlowChart.png)
